### PR TITLE
Add concept of adapters

### PR DESCRIPTION
**Pull request by nhunzaker**
This PR adds two new features:

1. `query`, `body`, and `headers` will now be folded together. This allows you to set up defaults without them being blown away.
2. The concept of _adapters_. This means, eventually, it will be possible to use more than AJAX for API communication. Gangway is still HTTP heavy, but it's a step towards a long-term goal.

Addresses #15

### CODE REVIEW

File: src/configure.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+let defaults = {
+  adapter    : require('./adapters/ajax'),
+  body       : undefined,
+  params     : undefined,
+  headers    : {},
+  method     : 'GET',
```
Comment (ltk) [inline src/configure.js]: a lot of this configuration doesn't make sense out of the context of the adapter. should there be a key under which all adapter configuration resides?

**Comment by ltk** [comment]
it's hard to imagine what the next non-ajax or non-http adapter will look like, but it's at least a fun exploration. :+1:

**Comment by nhunzaker** [comment]
@ltk I had a couple of thoughts, like you could create an adapter for the location service in the browser, or something else.

File: src/configure.js (hunk 1 above)
Comment (nhunzaker) [inline src/configure.js]: Yeah, I agree. Like the adapter needs it's own defaults... It's worth thinking that through all the way before this merges.

**Comment by nhunzaker** [comment]
I'm going to close this for now... Needs more thought.

**Pull request closed without merging**